src/content-scripts/inject/cosmjs-offline-signer.ts

```diff
@@ -0,0 +1,123 @@
+import {
+  Coin,
+  encodeSecp256k1Signature,
+  OfflineSigner,
+  serializeSignDoc
+} from "@cosmjs/launchpad";
+import { AccountData, SignResponse } from "@cosmjs/launchpad/types/signer";
+import {
+  GetKeyMsg,
+  RequestSignMsg,
+  RequestTxBuilderConfigMsg
+} from "../../background/keyring";
+import { sendMessage } from "../../common/message/send";
+import { BACKGROUND_PORT } from "../../common/message/constant";
+import { StdSignDoc } from "@cosmjs/launchpad/types/encoding";
+import { feeFromString } from "../../background/keyring/utils";
+
+const Buffer = require("buffer/").Buffer;
+
+export class CosmJSOfflineSigner implements OfflineSigner {
+  constructor(private readonly chainId: string) {}
+
+  async getAccounts(): Promise<AccountData[]> {
+    const key = await sendMessage(BACKGROUND_PORT, new GetKeyMsg(this.chainId));
+
+    return [
+      {
+        address: key.bech32Address,
+        // Currently, only secp256k1 is supported.
+        algo: "secp256k1",
+        pubkey: Buffer.from(key.pubKeyHex)
```

Comment:
Here an `"hex"` argument is missing. Otherwise this does ASCII encoding:

```
> Buffer.from("aabbccdd")
<Buffer 61 61 62 62 63 63 64 64>
```

Reply:
Oh, I made a mistake. Thank you.